freqtrade/exchange/kraken.py

```diff
@@ -133,24 +133,31 @@ def fill_leverage_brackets(self):
             Assigns property _leverage_brackets to a dictionary of information about the leverage
             allowed on each pair
         """
-        # TODO-lev: Not sure if this works correctly for futures
         leverages = {}
-        for pair, market in self._api.load_markets().items():
-            info = market['info']
-            leverage_buy = info['leverage_buy']
-            leverage_sell = info['leverage_sell']
-            if len(info['leverage_buy']) > 0 or len(info['leverage_sell']) > 0:
-                if leverage_buy != leverage_sell:
-                    print(f"\033[91m The buy leverage != the sell leverage for {pair}."
-                          "please let freqtrade know because this has never happened before"
-                          )
-                    if max(leverage_buy) < max(leverage_sell):
-                        leverages[pair] = leverage_buy
+        try:
+            for pair, market in self._api.load_markets().items():
```

Comment:
Looking at this closer - i think you should use `self.markets.items()`.

it's keeping a cached version of the markets (including a refresh logic).